pyiron_base/generic/datacontainer.py

```diff
@@ -510,6 +510,43 @@ def get(self, key, default=None, create=False):
         else:
             return super().get(key, default=default)
 
+    def search_tree(self, key, fail=True):
```

Comment:
I'm thinking to just call it `search`, since in general the `DataContainer` docs don't talk about "trees".  If that sounds to general, we could go with 'recursive' or 'nested' which is more consistent with the normal terminology in the docs.

Reply:
sounds good. I'll do it.